radio/hf_model.py

```diff
@@ -12,35 +12,56 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from collections import namedtuple
-from typing import Optional
+from typing import Optional, List, Union
 
 from timm.models import VisionTransformer
 import torch
 from transformers import PretrainedConfig, PreTrainedModel
 
 
+from .common import RESOURCE_MAP, DEFAULT_VERSION
+
+# Force import of eradio_model in order to register it.
 from .eradio_model import eradio
 from .radio_model import create_model_from_args
-from .radio_model import RADIOModel as RADIOModelBase
+from .radio_model import RADIOModel as RADIOModelBase, Resolution
 from .input_conditioner import get_default_conditioner, InputConditioner
+
+
 # Register extra models
 from .extra_timm_models import *
 
+
 class RADIOConfig(PretrainedConfig):
     """Pretrained Hugging Face configuration for RADIO models."""
 
     def __init__(
         self,
         args: Optional[dict] = None,
-        version: Optional[str] = "v1",
-        return_summary: Optional[bool] = True,
-        return_spatial_features: Optional[bool] = True,
+        version: Optional[str] = DEFAULT_VERSION,
+        patch_size: Optional[int] = None,
+        max_resolution: Optional[int] = None,
```

Comment:
Forgive my ignorance: Are these considered part of the config, or innately part of the model? It shouldn't be user configurable, but is this config allowing for that to be specified by the user?

Reply:
This is part of the config as we need it to instantiate the `RADIOModel`. We don't expect users to change it as the standard programming model fetches the config seamlessly:

```python
model = AutoModel.from_pretrained(hf_repo, trust_remote_code=True)
```